server/channels/app/channel_bookmark_test.go

```diff
@@ -102,8 +102,12 @@ func TestUpdateBookmark(t *testing.T) {
 			HasPreviewImage: true,
 		}
 
-		th.App.Srv().Store().FileInfo().Save(th.Context, file)
-		defer th.App.Srv().Store().FileInfo().PermanentDelete(th.Context, file.Id)
+		_, err := th.App.Srv().Store().FileInfo().Save(th.Context, file)
+		assert.NoError(t, err)
+		defer func() {
+			err = th.App.Srv().Store().FileInfo().PermanentDelete(th.Context, file.Id)
+			assert.NoError(t, err)
+		}()
 
 		bookmark2 := createBookmark("File to be updated", model.ChannelBookmarkFile, th.BasicChannel.Id, file.Id)
 		bookmarkResp, err := th.App.CreateChannelBookmark(th.Context, bookmark2, "")
```

Comment:
We are now getting a conflict between `error` and `*model.AppError`. The clean solution would be to rename this `err` variable to `appErr`.

Reply:
Done!

Reply:
The issue isn't fixed yet. Do you need any help?

Reply:
This is still a concern. See the CI error:

```

=== FAIL: channels/app TestUpdateBookmark/same_user_update_a_channel_bookmark (re-run 1) (0.01s)
    channel_bookmark_test.go:114: 
        	Error Trace:	/mattermost/server/channels/app/channel_bookmark_test.go:114
        	            				/mattermost/server/channels/app/channel_bookmark_test.go:168
        	Error:      	Received unexpected error:
        	            	<nil>
        	Test:       	TestUpdateBookmark/same_user_update_a_channel_bookmark
```

Reply:
@RS-labhub Do you have any questions regarding the feedback?

Reply:
@hanzei check this: https://github.com/mattermost/mattermost/pull/28789#issuecomment-2468009428

Reply:
You are missing up the `error` type and the `*model.AppError` type. This line needs to become a new variable as `err` is of type `error`.